assets/src/hooks/useChannel.ts

```diff
@@ -0,0 +1,64 @@
+import { Channel, Socket } from "phoenix"
+import { useEffect, useState } from "react"
+import { reload } from "../models/browser"
+
+/** Opens a channel for the given topic
+ *  and returns the latest data that's been pushed to it
+ *
+ *  Only listens for a single event
+ *  Assumes the data returned on join and the data from subsequent pushes are the same format
+ *  If topic is null, does not open a channel.
+ *  Returns offState if the channel is not open yet
+ *  Returns loadingState after the channel has been opened but before receiving the first response
+ */
+export const useChannel = <T>({
+  socket,
+  topic,
+  event,
+  parser,
+  loadingState,
+  offState,
+}: {
+  socket: Socket | undefined
+  topic: string | null
+  event: string
+  parser: (data: any) => T
+  loadingState: T
+  offState: T
+}): T => {
+  const [result, setResult] = useState<T>(offState)
```

Comment:
Nit: To me, `result` makes me think of a single response. Would something like `state` convey what this is better?

Reply:
This is a single response. There's a simple `return result` at the end of the hook.

(The channel isn't saved by a `setState`.)

Reply:
Sorry, I phrased that poorly. What I meant was more like a single value and the interaction is over, as opposed to the current value of some data that will be updated over time.